lib/apiGateway/schema.js

```diff
@@ -119,6 +129,71 @@ const partitionKey = Joi.alternatives().try([
     )
 ])
 
+const allowedDynamodbActions = ['PutItem', 'GetItem', 'DeleteItem', 'UpdateItem']
+const dynamodbDefaultKeyScheme = Joi.alternatives().try([
+  Joi.string(),
+  Joi.object()
+    .keys({
+      pathParam: Joi.string(),
+      queryStringParam: Joi.string(),
+      attributeType: Joi.string().required()
+    })
+    .xor('pathParam', 'queryStringParam')
+    .error(
+      customErrorBuilder(
+        'object.xor',
+        'key must contain "pathParam" or "queryStringParam" and only one'
```

Comment:
this reads a bit odd, maybe `key must contain "pathParam" or "queryStringParam", but not both`?